src/endpoints/getCredential.ts

```diff
@@ -5,83 +5,94 @@ import jwt from 'jsonwebtoken';
-        return res.sendStatus(401);
-    }
-
-    try {
-        const credential = await payload.findByID({
-            id,
-            collection: 'credential',
-            depth: 3,
-        });
+    id = decoded.id;
+  } catch (error) {
+    return res.sendStatus(401);
+  }
 
-        if (credential.status === 'REVOKED') {
-            return res.status(410).json({ reason: 'Credential has been revoked' });
-        }
+  try {
+    const credential = await payload.findByID({
+      id,
+      collection: 'credential',
+      depth: 3,
+    });
 
-        if (
-            typeof credential?.batch === 'string' ||
-            typeof credential.batch.template === 'string' ||
-            !credential.batch.template.credentialTemplateJson
-        ) {
-            return res.sendStatus(404);
-        }
+    if (credential.status === 'REVOKED') {
+      return res.status(410).json({ reason: 'Credential has been revoked' });
+    }
 
-        const builtCredential = insertValuesIntoHandlebarsJsonTemplate(
-            JSON.stringify(credential.batch.template.credentialTemplateJson),
-            {
-                ...(credential.extraFields as any),
-                credentialName: credential.credentialName,
-                earnerName: credential.earnerName,
-                emailAddress: credential.emailAddress,
-                now: new Date().toISOString(),
-                issuanceDate: credential.updatedAt,
-            }
-        ) as any as UnsignedVC;
-
-        if (typeof builtCredential?.issuer === 'string') builtCredential.issuer = {};
-        if ('id' in (builtCredential?.issuer ?? {})) delete builtCredential.issuer.id;
-
-        res.status(200).json(builtCredential);
-    } catch (error) {
-        console.error(error);
-        res.sendStatus(500);
+    if (
+      typeof credential?.batch === 'string' ||
+      typeof credential.batch.template === 'string' ||
+      !credential.batch.template.credentialTemplateJson
+    ) {
+      return res.sendStatus(404);
     }
+
+    const builtCredential = insertValuesIntoHandlebarsJsonTemplate(
+      JSON.stringify(credential.batch.template.credentialTemplateJson),
+      {
+        ...(credential.extraFields as any),
+        credentialName: credential.credentialName,
+        earnerName: credential.earnerName,
+        emailAddress: credential.emailAddress,
+        now: new Date().toISOString(),
+        issuanceDate: credential.updatedAt,
+      }
+    ) as any as UnsignedVC;
+
+    builtCredential.id = `urn:uuid:${id}`;
+
+    if (typeof builtCredential?.issuer === 'string') builtCredential.issuer = {};
+    if ('id' in (builtCredential?.issuer ?? {})) delete builtCredential.issuer.id;
+
+    res.status(200).json(
+      {
+        ...builtCredential,
+        extra: {
+          credentialName: credential.credentialName,
+          earnerName: credential.earnerName,
+          awardedDate: credential.updatedAt,
+          issuedDate: new Date().toISOString(),
+        }
+      });
```

Comment:
Putting the extra fields into the credential itself seems not quite right -- there is a remote danger of overwriting some field in the credential.

Since this is effectively a new endpoint, couldn't we return a structure `{credential, extra}`?